### PR TITLE
[MXNET-116] Optimized functions with batch input

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
Improved the ImageClassifier and ObjectDetector to receive and process batch inputs. It will help users to manager the memory usage and speed up the inference
@nswamy @Roshrini 
### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: scala-package/infer/src/main/scala/ml/dmlc/mxnet/infer/ImageClassifier.scala [hunk 1]
```diff
@@ -195,19 +195,28 @@ object ImageClassifier {
...
-    * @return List of buffered images
+    * @param inputImageFiles
+    * @return List of buffered images-
     */
-  def loadInputBatch(inputImageDirPath: String): List[BufferedImage] = {
+  def loadInputBatch(inputImageFiles: List[File]): List[BufferedImage] = {
+    inputImageFiles.map(file => ImageIO.read(file))
+  }
+
+  def generateBatches(inputImageDirPath: String, batchSize: Int = 100): List[List[File]] = {
```
Comment (nswamy) [inline scala-package/infer/src/main/scala/ml/dmlc/mxnet/infer/ImageClassifier.scala]: don't feel like this belongs in ImageClassifier, please move to the example

File: scala-package/infer/src/main/scala/ml/dmlc/mxnet/infer/ImageClassifier.scala [hunk 2]
```diff
@@ -195,19 +195,28 @@ object ImageClassifier {
 
   /**
     * Loading input batch of images
-    * @param inputImageDirPath
-    * @return List of buffered images
+    * @param inputImageFiles
+    * @return List of buffered images-
     */
-  def loadInputBatch(inputImageDirPath: String): List[BufferedImage] = {
+  def loadInputBatch(inputImageFiles: List[File]): List[BufferedImage] = {
```
Comment (nswamy) [inline scala-package/infer/src/main/scala/ml/dmlc/mxnet/infer/ImageClassifier.scala]: take path to files and return a traversable of Buffered Image

check the file exists in the function

**Comment by lanking520** [comment]
@marcoabreu Could you please take a look in here. This is an updated PR for testing the APIs on GPU, but it seemed the CI has not sync with the most recent change from master branch. Cannot see Scala GPU in the test

**Comment by marcoabreu** [comment]
@lanking520 Scala GPU is there: http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/PR-10343/7/pipeline/597

**Pull request merged**